### PR TITLE
(feat:949): Enabled home microservice to push data for for search endpoint.

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 949

# Explain the feature

- Enabled the home microservice to push the data to search. When ever an creation/updation/deletion happens on the records. this will automate the search index to get updated with data.

## Does this PR introduce a breaking change
No

## Screenshots

N/A
### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/search-spa/src/assets/icons/home.svg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0"?>
```
Comment (deshmukhmayur) [inline packages/search-spa/src/assets/icons/home.svg:1]: Can you remove the icon from the service? Add it to the global assets, and use absolute path of that for indexing if needed.

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

**Review by ghanlohar — APPROVED** [review]

File: packages/search-spa/src/assets/icons/home.svg (hunk 1 above)
Comment (riginoommen) [reply packages/search-spa/src/assets/icons/home.svg]: Removed asset and updating the same is asset spa

File: packages/home-service/.env.example [hunk 2]
```diff
@@ -15,3 +15,10 @@ USER_SERVICE_SERVICE_PORT=
 ## Redis Host
 REDIS_SERVICE_HOST=redis
 # REDIS_SERVICE_PORT=
+
+## API Gateway
+API_GATEWAY=
+GATEWAY_AUTH_TOKEN=
+
+## URLs
+CLIENT_URL=
```
Comment (deshmukhmayur) [inline packages/home-service/.env.example:24]:
```suggestion
CLIENT_URL=

```

Newline at the end of the file.

File: packages/home-service/src/helpers.ts [hunk 3]
```diff
@@ -94,6 +95,75 @@ class HomeAPIHelper {
...
+    return {
+      'input': {
+        'dataSource': 'oneportal',
+        'documents': {
+          'id': `${data._id}`,
+          'title': data.name,
+          'abstract': data.description || '',
+          'description': data.description || '',
+          'icon': `assets/icons/home.svg`,
+          'uri': `${process.env.CLIENT_URL + data.link}`,
```
Comment (deshmukhmayur) [inline packages/home-service/src/helpers.ts:110]:
```suggestion
          'uri': process.env.CLIENT_URL + data.link,
```

File: packages/home-service/src/helpers.ts [hunk 4]
```diff
@@ -94,6 +95,75 @@ class HomeAPIHelper {
...
     } );
   }
+
+  // Search Data formatter
+  public formatSearchInput(data: any) {
+    return {
+      'input': {
+        'dataSource': 'oneportal',
+        'documents': {
+          'id': `${data._id}`,
```
Comment (deshmukhmayur) [inline packages/home-service/src/helpers.ts:105]:
```suggestion
          'id': data._id,
```

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]
As per our discussion today, should we still include the updateHomeIndex?

I would suggest to keep a generic endpoint for updateIndex with a pre-defined input type, which other apps can use to update the index.

**Review by deshmukhmayur — APPROVED** [review]

**Pull request merged**